### PR TITLE
Correctly handle vendored updates in nested directory

### PR DESCRIPTION
**Pull request by jurre**
Previously we would prepend the nested directory to the filename, but
this is normally stored separately in the `directory` field of a
DependencyFile.

This would result in PRs being opened with the vendored files in one
directory too deep (`custom_dir/custom_dir/vendor` instead of
`custom_dir/vendor`).

Fixes: https://github.com/dependabot/dependabot-core/issues/2687

### CODE REVIEW

File: common/lib/dependabot/file_updaters/vendor_updater.rb [hunk 1]
```diff
@@ -31,8 +31,10 @@ def updated_vendor_cache_files(base_directory:)
               encoding = Dependabot::DependencyFile::ContentEncoding::BASE64
               encoded_content = Base64.encode64(encoded_content) unless deleted
             end
+
+            clean_base_directory = Pathname.new(base_directory).basename
```
Comment (feelepxyz) [inline common/lib/dependabot/file_updaters/vendor_updater.rb:35]: Is `basename` the right thing here? Looks like it just grabs the last component `Pathname.new("dir/nested").basename` > `nested` 

Should it instead be something like?

```
project_root  = Pathname.new(File.expand_path(File.join(Dir.pwd, base_directory)))
filename = Pathname.new(path).expand_path.relative_path_from(project_root)

Comment (jurre) [reply common/lib/dependabot/file_updaters/vendor_updater.rb]: Yeah great suggestion, I've improved the other relative path code as well

File: common/lib/dependabot/file_updaters/vendor_updater.rb [hunk 2]
```diff
@@ -18,7 +18,8 @@ def updated_vendor_cache_files(base_directory:)
         return [] unless repo_contents_path && vendor_dir
 
         Dir.chdir(repo_contents_path) do
-          relative_dir = vendor_dir.sub("#{repo_contents_path}/", "")
+          relative_dir = Pathname.new(vendor_dir).relative_path_from(Dir.pwd)
```
Comment (feelepxyz) [inline common/lib/dependabot/file_updaters/vendor_updater.rb:21]: 👍

**Review by feelepxyz — APPROVED** [review]
Nice one! LGTM 👍

**Pull request merged**